### ISSUE TITLE
Update JSON payload schema and API endpoint names

### ISSUE THREAD

**Issue by JohnGiorgi** [PathwayCommons/semantic-search#65]
# Overview

After discussion, we have agreed to standardize the format of the JSON payload of the `POST` request and update the endpoints. 

## JSON Payload

My proposal is to switch from supporting two styles of input to just supporting the following style:

```json
{
    "query": {
        "uid":"9887103",
        "text": "The Drosophila activin..."
    },
    "documents":[
        {
            "uid": "9887103",
            "text": "The Drosophila activin .."
        },
        {
            "uid": "30049242",
            "text": "Transcriptional up-regulation..."
        },
        {
            "uid": "22936248",
            "text": "High-fidelity..."
        }
    ],
    "top_k":2
}
```

where the `"text"` field is optional. When not provided, we assume the `"uid"` is a valid PMID and use the `ncbi.py` module to fetch the text.  This is a very small change to what we currently do. It basically amounts to dropping support for the other style of query, which looked like:

```json
{
    "query": "9887103",
    "documents": ["9887103", "30049242", "22936248"],
    "top_k": 2
}
```

## API Endpoints

Again, I propose a simple change here. Let's use the route `"/"` to be a health-check, which I believe is a pretty standard convention. All it will do is return a status code of `200` (and maybe a message, like `"Hello world!"`, or `"I'm alive!"`).

The current route `"/"` will be replaced with a new route, perhaps `"/query-with-uids"`. We can debate the name, but it should be descriptive and follow a naming pattern that we can extend in the future (e.g. `"/query-with-text"` or w/e).

## TODO

@maxkfranz, @jvwong and @Anwesh1, please let me know what you think about this proposal. I am happy to hear suggestions (a different JSON payload format, a different name for the endpoint, etc). 

Once we are happy, updating the code should be straightforward. I think it makes sense to pause and do this now, before we add any additional features.

**Comment by JohnGiorgi** [comment]
@maxkfranz Sorry Max, I believe you are off work for a little while, disregard!

**Comment by jvwong** [comment]
I think it's fine. 

To be totally unambiguous, you could have a field to identify the database collection (e.g. [`pubmed`](https://registry.identifiers.org/registry/pubmed):

```json
{
  "dbPrefix": "pubmed",
  "query": 
...
}
```

**Comment by JohnGiorgi** [comment]
I see, yeah that makes sense. The only edge case I can think of is when you supply documents with ids from different databases:

```json
{
    "query": {
        "uid":"I'm a PMID!",
    },
    "documents":[
        {
            "uid": "I'm a Semantic Scholar ID!",
        },
    ],
}
```

in which case `"dbPrefix"` would have to be an argument alongside `"uid"` and `"text"`:

```json
{
    "query": {
        "uid":"I'm a PMID!",
        "dbPrefix": "pubmed"
    },
    "documents":[
        {
            "uid": "I'm a Semantic Scholar ID!",
            "dbPrefix": "semantic_scholar"
        },
    ],
}
```

but that gets a little verbose. So, we could add `"dbPrefix"` as a top-level argument and assume all `"uid"`s are from this db.

**Comment by Anwesh1** [comment]
I think this is good to implement? Makes sense to keep the `"dbPrefix"` a top-level argument or it might make the structure a bit more cluttered.

**Comment by maxkfranz** [comment]
> I think this is good to implement. Makes sense to keep the `"dbPrefix"` a top-level argument or it might make the structure a bit more cluttered.

A simpler alternative may be for the IDs to be unique as a precondition, e.g. `somedb:someid`.

**Comment by JohnGiorgi** [comment]
Closes by #68.
